src/site/content/en/payments/registering-a-web-based-payment-app/index.md

```diff
@@ -0,0 +1,321 @@
+---
+layout: post
+title: Registering a web-based payment app
+subhead: |
+  Learn how to configure a web-based payment app on registration.
+authors:
+  - agektmr
+date: 2020-07-17
+description: |
+  Learn how to register a web-based payment app to a customers' browser. You'll also learn how to debug them.
+tags:
+  - payments
+---
+
+Web-based payment apps are [Progressive Web Apps
+(PWA)](/progressive-web-apps/) and run on top of [service
+workers](https://developer.mozilla.org/docs/Web/API/Service_Worker_API). The
+service worker in a payment app plays an important role as it captures payment
+requests from a merchant, launches the payment app, and mediates the
+communication with the merchant.
+
+To configure a web-based payment app, you need to register available payment
+methods, delegation options, and a service worker. You can configure your
+web-based payment app declaratively with a web app manifest or imperatively with
+JavaScript.
+
+## Configuring a payment app with a web app manifest
+
+To configure your web-based payment app declaratively, [serve a web app
+manifest](/setting-up-a-payment-method/#step-3:-serve-a-web-app-manifest).
+
+The following properties in the web app manifest are relevant for web-based payment apps:
+* `name`
+* `icons`
+* `serviceworker`
+    * `src`
+    * `scope`
+    * `use_cache`
+* `payment`
+    * `supported_delegations`
+
+Check out [Setting up a payment
+method](/setting-up-a-payment-method/#step-3:-serve-a-web-app-manifest)
+to make sure your payment method manifest points to your web app manifest
+properly.
+
+### Registering a service worker just-in-time (JIT)
+
+Service workers are usually registered using JavaScript, but they can also be
+automatically registered by the browser when the user chooses to pay with a
+web-based payment app on a merchant website. This is called "just-in-time (JIT)
+registration".
+
+The JIT registration requires only serving [the web app
+manifest](/setting-up-a-payment-method/#step-3:-serve-a-web-app-manifest)
+and no additional coding. If you've already configured your web app manifest and
+are serving it properly, you should be all set. The browser will handle the
+rest.
+
+{% Aside 'caution' %}
+While a single payment method identifier can support multiple payment apps,
+JIT-registration happens only when the payment method manifest points to a
+single payment app.
+{% endAside %}
+
+## Configure a payment app with JavaScript
+
+Configuring a web-based payment app using JavaScript allows you to register
+multiple payment methods and manually unregister service workers.
+
+### Manually register a service worker
```

Comment:
Let's recommend using Workbox somewhere in this section?

Maybe we can use this include: https://github.com/GoogleChrome/web.dev/blob/master/src/site/_includes/content/reliable/workbox.njk

You would use this code to add the include: `{% include 'content/reliable/workbox.njk' %}`

Reply:
Why? Workbox doesn't have anything that helps payment handler?

Reply:
If someone only uses a service worker to handle Payment Handler stuff, then I agree, Workbox is unnecessary.

However, I'm worried that once we make our audience aware of service workers, they will realize that they can use it for other tasks, such as caching. In that case, we absolutely want them to consider Workbox first.

So, how about we rephrase the section along these general lines:

* If you're **only** using a service worker for basic Payment Handler logic, then you can manually implement one yourself.
* If you start using your service worker for other use cases, such as caching, we strongly recommend using Workbox as the foundation of your service worker.

cc @jeffposnick